jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcDataContext.java

```diff
@@ -336,6 +336,13 @@ public DataSet executeQuery(CompiledQuery compiledQuery, Object... values) {
 
     private DataSet execute(Connection connection, Query query, Statement statement, JdbcCompiledQuery compiledQuery,
             JdbcCompiledQueryLease lease, Object[] values) throws SQLException, MetaModelException {
+        Integer maxRows = query.getMaxRows();
+
+        final List<SelectItem> selectItems = query.getSelectClause().getItems();
+        if (maxRows != null && maxRows.intValue() == 0) {
```

Comment:
While that is true, I really don't hope anyone would go "Oh, I'll remove that" without at least thinking about why that would be there.

Anyway, it's not really an issue.